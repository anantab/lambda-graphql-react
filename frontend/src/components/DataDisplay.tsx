import * as React from 'react';
import { formatDate, formatTime } from '../lib';

class DataDisplay extends React.Component<any, any> {
    public render() {
        return (
            <table className="table table-bordered">
                <tbody>
                    <tr>
                        <td colSpan={2}>{formatDate(this.props.item.date)}</td>
                    </tr>
                    <tr>
                        <td colSpan={2} className="table-secondary">{this.props.item.currency}</td>
                    </tr>
                    <tr>
                        <td><strong>Buy</strong></td>
                        <td><strong>Sell</strong></td>
                    </tr>
                    <tr>
                        <td>${this.props.item.buy_price}</td>
                        <td>${this.props.item.sell_price}</td>
                    </tr>
                    <tr>
                        <td>{formatTime(this.props.item.buy_time)}</td>
                        <td>{formatTime(this.props.item.sell_time)}</td>
                    </tr>
                    <tr>
                        <td colSpan={2}>Profit: ${this.props.item.profit}</td>
                    </tr>
                </tbody>
            </table>
        );
    }
}

export default DataDisplay;

