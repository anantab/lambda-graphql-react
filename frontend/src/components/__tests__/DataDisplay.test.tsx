import * as React from 'react';
import * as renderer from 'react-test-renderer';
import DataDisplay from '../DataDisplay';
test('Renders data correctly', () => {
    const item = {
        "date": "20180507",
        "currency": "BTC",
        "buy_time": "0915",
        "buy_price": "34.98",
        "sell_time": "1230",
        "sell_price": "37.01",
        "profit": "2.03"
    }

    const component = renderer.create(
        <DataDisplay item={item} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});