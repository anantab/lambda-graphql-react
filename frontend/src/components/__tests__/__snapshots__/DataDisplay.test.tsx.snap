// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders data correctly 1`] = `
<table
  className="table table-bordered"
>
  <tbody>
    <tr>
      <td
        colSpan={2}
      >
        07-May-18
      </td>
    </tr>
    <tr>
      <td
        className="table-secondary"
        colSpan={2}
      >
        BTC
      </td>
    </tr>
    <tr>
      <td>
        <strong>
          Buy
        </strong>
      </td>
      <td>
        <strong>
          Sell
        </strong>
      </td>
    </tr>
    <tr>
      <td>
        $
        34.98
      </td>
      <td>
        $
        37.01
      </td>
    </tr>
    <tr>
      <td>
        9:15AM
      </td>
      <td>
        12:30PM
      </td>
    </tr>
    <tr>
      <td
        colSpan={2}
      >
        Profit: $
        2.03
      </td>
    </tr>
  </tbody>
</table>
`;
